Formula/c/charmcraft.rb

```diff
@@ -17,20 +17,12 @@ class Charmcraft < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "7d6ce9798ec325abab4604c95e8754ce5872e2563b22cd4e43a4560c4cbae358"
   end
 
-  depends_on "rust" => :build # for rpds-py
-  depends_on "cffi"
-  depends_on "keyring"
```

Comment:
May need some extra work when removing `keyring` because the dependency trees are different between macOS and Linux. Automatic resource updates will only add the ones from local OS.

https://github.com/jaraco/keyring/blob/main/setup.cfg#L22-L23

I think the easiest workaround is to add extra Linux resources to `extra_packages`.